tests/test_robot.py

```diff
@@ -18,36 +15,33 @@ def setUp(self):
         ]
 
     def test_no_warning(self):
-        with open('tests/test_in/junit_no_fail.xml', 'r') as xmlfile:
+        with open('tests/test_in/junit_no_fail.xml') as xmlfile:
             self.warnings.check(xmlfile.read())
         self.assertEqual(self.warnings.return_count(), 0)
 
     def test_single_warning(self):
-        with open('tests/test_in/robot_single_fail.xml', 'r') as xmlfile:
-            with patch('sys.stdout', new=StringIO()) as fake_out:
+        with open('tests/test_in/robot_single_fail.xml') as xmlfile:
+            with self.assertLogs(level="INFO") as fake_out:
                 self.warnings.check(xmlfile.read())
                 count = self.warnings.return_count()
-        stdout_log = fake_out.getvalue()
+        stdout_log = fake_out.output
 
         self.assertEqual(count, 1)
-        self.assertIn("Suite {!r}: 1 warnings found".format(self.suite1), stdout_log)
-        self.assertIn("Suite {!r}: 0 warnings found".format(self.suite2), stdout_log)
+        self.assertIn("INFO:root:Suite One &amp; Suite Two.Suite One.First Test", stdout_log)
 
     def test_double_warning_and_verbosity(self):
-        with open('tests/test_in/robot_double_fail.xml', 'r') as xmlfile:
-            with patch('sys.stdout', new=StringIO()) as fake_out:
+        with open('tests/test_in/robot_double_fail.xml') as xmlfile:
+            with self.assertLogs(level="INFO") as fake_out:
                 self.warnings.check(xmlfile.read())
                 count = self.warnings.return_count()
-        stdout_log = fake_out.getvalue()
+        stdout_log = fake_out.output
 
         self.assertEqual(count, 2)
         self.assertEqual(
-            '\n'.join([
-                "Suite One &amp; Suite Two.Suite One.First Test",
-                "Suite One &amp; Suite Two.Suite Two.Another test",
-                "Suite {!r}: 1 warnings found".format(self.suite1),
-                "Suite {!r}: 1 warnings found".format(self.suite2),
-            ]) + '\n',
+            [
+                "INFO:root:Suite One &amp; Suite Two.Suite One.First Test",
+                "INFO:root:Suite One &amp; Suite Two.Suite Two.Another test",
+            ],
             stdout_log
```

Comment:
Please use `fake_out.output` directly.

Is there information loss? These logs don't tell me much. This test case should test the reporting of the amount of warnings.

Reply:
Resolved in 93864a7c4902c5ef4685e11b2d27aa0b2d4d6ece